### PR TITLE
Canavandl/svg headless export

### PR DESCRIPTION
**Pull request by canavandl**
- [x] issues: related #6345 
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

The main point of discussion is that I have this functionality as a separate function than ``bokeh.io.export``. Is that the correct idea? I think it is because A) users would potentially want a PNG screenshot of an SVG plot-containing layout and B) the product is different enough (a single PNG vs multiple SVGs) that I think it'd cause confusion if they were the same method. Perhaps we'd better rename ``export`` to ``export_as_png``?

### CODE REVIEW

File: bokeh/io.py [hunk 1]
```diff
@@ -658,3 +664,75 @@ def export(obj, filename=None):
...
+    return serialized_svgs
+    """
+
+    svgs = driver.execute_script(svg_script)
+
+    driver.quit()
+
+    return svgs
+
+def export_svgs(obj, filename=None):
```
Comment (canavandl) [inline bokeh/io.py]: maybe ``export_as_svgs``?

File: bokeh/io.py [hunk 2]
```diff
@@ -658,3 +664,75 @@ def export(obj, filename=None):
...
+    save(obj, filename=html_path, resources=INLINE, title="")
+
+    driver = webdriver.PhantomJS()
+    driver.get("file:///" + html_path)
+
+    _wait_until_render_complete(driver)
+
+    svg_script = """
+    var serialized_svgs = [];
+    var svgs = document.getElementsByClassName('bk-root')[0].getElementsByTagName("svg");
```
Comment (canavandl) [inline bokeh/io.py]: there's never going to be a case where a SVG is under the ``bk-root`` node, is there?

File: bokeh/io.py [hunk 3]
```diff
@@ -658,3 +664,75 @@ def export(obj, filename=None):
...
+    svgs = _get_svgs(obj)
+
+    if len(svgs) == 0:
+        logger.warn("No SVG Plots were found.")
+        return
+
+    if filename is None:
+        filename = _detect_filename("svg")
+
+    for i, svg in enumerate(svgs):
```
Comment (canavandl) [inline bokeh/io.py]: this seems brittle

File: bokeh/io.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/io.py]: I don't have a preference

File: bokeh/io.py (hunk 2 above)
Comment (bryevdv) [inline bokeh/io.py]: Will `ImageURL` load SVGs? That's the only possibility I can think of without the presence of custom extensions. I think this is fine for now in any case, if there are corner cases later we can deal with them as they arise.

File: bokeh/io.py (hunk 3 above)
Comment (bryevdv) [inline bokeh/io.py]: More tests are always welcome but not sure what specifically you are referring to as brittle here

File: bokeh/io.py (hunk 1 above)
Comment (canavandl) [inline bokeh/io.py]: I think it's going to be ``export_svgs`` and renaming ``export`` -> ``export_png`` for clarity.

File: bokeh/io.py (hunk 2 above)
Comment (canavandl) [inline bokeh/io.py]: ImageURL doesn't work w/ SVGs, so I think we're fine.

**Comment by bryevdv** [comment]
@canavandl last CI error is just a docs ref issue:
```
Traceback (most recent call last):
  File "/home/travis/miniconda/lib/python3.4/site-packages/Sphinx-1.5.1-py3.4.egg/sphinx/cmdline.py", line 296, in main
    app.build(opts.force_all, filenames)
  File "/home/travis/miniconda/lib/python3.4/site-packages/Sphinx-1.5.1-py3.4.egg/sphinx/application.py", line 333, in build
    self.builder.build_update()
  File "/home/travis/miniconda/lib/python3.4/site-packages/Sphinx-1.5.1-py3.4.egg/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/home/travis/miniconda/lib/python3.4/site-packages/Sphinx-1.5.1-py3.4.egg/sphinx/builders/__init__.py", line 268, in build
    self.warn(*warning, **kwargs)
  File "/home/travis/miniconda/lib/python3.4/site-packages/Sphinx-1.5.1-py3.4.egg/sphinx/application.py", line 399, in warn
    raise SphinxWarning(warntext)
sphinx.errors.SphinxWarning: /home/travis/build/bokeh/bokeh/sphinx/source/docs/user_guide/export.rst:25: ERROR: Undefined substitution referenced: "export_png".
Warning, treated as error:
/home/travis/build/bokeh/bokeh/sphinx/source/docs/user_guide/export.rst:25: ERROR: Undefined substitution referenced: "export_png".
```

**Comment by bryevdv** [comment]
@canavandl I just edited the file here in the GH UI, seemed simple enough that I couldn't botch it :) Will  merge when green.

**Comment by bryevdv** [comment]
> that I couldn't botch it

ha-ha
